tls/s2n_server_key_exchange.c

```diff
@@ -100,7 +100,7 @@ int s2n_ecdhe_server_key_recv_read_data(struct s2n_connection *conn, struct s2n_
 
 int s2n_ecdhe_server_key_recv_parse_data(struct s2n_connection *conn, struct s2n_kex_raw_server_data *raw_server_data)
 {
-    POSIX_GUARD(s2n_ecc_evp_parse_params(&raw_server_data->ecdhe_data, &conn->kex_params.server_ecc_evp_params));
+    POSIX_GUARD(s2n_ecc_evp_parse_params(conn, &raw_server_data->ecdhe_data, &conn->kex_params.server_ecc_evp_params));
 
```

Comment:
Do you need to check that the curve is allowed by the security policy in s2n_ecc_evp_parse_params, or can you do it here?

Reply:
I can do it in either functions. I placed that logic in `s2n_ecc_evp_parse_params` because it already had logic which checked if a curve was in some the all_curves list